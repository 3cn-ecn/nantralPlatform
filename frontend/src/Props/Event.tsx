import { snakeToCamelCase } from '../utils/camel';

export interface EventProps {
  id: number;
  color: string | null; // Color of the event card
  beginDate: Date; // Event begin date in Date format
  endDate: Date; // Event begin date in Date format
  description: string; // Description of the event in html
  getAbsoluteUrl: string; // url of the event from root url
  groupSlug: string; // slug of the organiser
  groupName: string; // name of the organiser
  image: string | null; // uri of the banner
  isMember: boolean;
  isParticipating: boolean; // whether connected user particpate at this event
  location: string; // where the event is taking place
  numberOfParticipants: number;
  publicity: 'Pub';
  slug: string; // slug of the event
  title: string;
  maxParticipant: number | null; // number max of participant
  endInscription: string | null; // date of the end of inscription
  beginInscription: string | null; // date of the beginning of inscription
  formUrl: string | null; // url of inscription form
  isFavorite: boolean;
  effectiveSize?: number; // The number of horizontal parts the events will used
  globalSize?: number; // The number of parts available
  position?: number; // The position in number of parts in the calendar
  placed?: boolean; // whether the event has been placed in the calendar
}

export function eventsToCamelCase(events: Array<any>) {
  events.forEach((event) => {
    // delete when date update to beginDate
    event.begin_date = event.date;

    // delete when endDate defined forEach event
    if (event.end_date === null) {
      event.end_date = new Date(new Date(event.date).getTime() + 3600000);
    }

    snakeToCamelCase(event, { beginDate: 'Date', endDate: 'Date' });
  });
}
