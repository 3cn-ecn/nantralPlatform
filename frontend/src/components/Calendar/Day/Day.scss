.dayData{
  height: 1.2rem;
   margin: 0.2rem;
}