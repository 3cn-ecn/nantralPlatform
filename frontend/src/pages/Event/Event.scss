@import './../../index.scss';


// .ListButton {
//   background-color: $primary-red-color;;
//   width: 100px;
//   height: 20px;
// }


// .CalendarButton {
//   background-color: $primary-red-color;;
//   width: 100px;
//   height: 20px;
// }

