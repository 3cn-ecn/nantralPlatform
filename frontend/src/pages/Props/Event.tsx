export interface EventProps {
  color: string | null; // Color of the event card
  date: Date; // Event begin date
  description: string; // Description of the event in html
  get_absolute_url: string; // url of the event from root url
  get_group_name: string; // name of the organiser
  group: string; // slug of the organiser
  image: string | null; // uri of the banner
  is_member: boolean;
  is_participating: boolean; // whether connected user particpate at this event
  location: string; // where the event is taking place
  number_of_participants: number;
  publicity: 'Pub';
  slug: string; // slug of the event
  title: string;
  max_participant: number | null; // number max of participant
  end_inscription: Date | null; // date of the end of inscription
  begin_inscripion: Date | null; // date of the beginning of inscription
  end_date: Date | null; // end date of the event
  ticketing: string | null; // url of inscription form
  effectiveSize?: number;
  globalSize?: number;
}
