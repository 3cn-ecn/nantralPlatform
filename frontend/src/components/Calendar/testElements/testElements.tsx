import { EventProps } from '../../../Props/Event';

export function createTestEvent() {
  const event: EventProps = {
    color: 'blue',
    beginDate: new Date('2023-02-17T03:24:00'),
    endDate: new Date('2023-02-17T04:24:00'),
    description: 'This is an event to test',
    getAbsoluteUrl: 'https://nantral-platform.fr/account/login/?next=/',
    getGroupName: 'Moi-mÃªme',
    group: 'Mon groupe',
    image: null,
    isMember: true,
    isParticipating: true,
    location: 'My home',
    numberOfParticipants: 15,
    publicity: 'Pub',
    slug: 'The slug',
    title: 'The test event',
    maxParticipant: null,
    endInscription: null,
    beginInscription: null,
    ticketing: null,
    isFavorite: true,
    effectiveSize: 1,
    globalSize: 1,
    position: 0,
    placed: false,
  };

  return event;
}
