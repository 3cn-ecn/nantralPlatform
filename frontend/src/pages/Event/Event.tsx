/* eslint-disable prettier/prettier */
import React from 'react';
import { Box, Tab } from '@mui/material';
import { TabContext, TabList, TabPanel } from '@mui/lab';
import './Event.scss';
import axios from 'axios';

import {
  CalendarMonth,
  CalendarToday,
  CalendarViewDay,
} from '@mui/icons-material';
import { EventProps, eventsToCamelCase } from '../../Props/Event';
import FilterBar from '../../components/FilterBar/FilterBar';
import Calendar from '../../components/Calendar/Calendar';
import Formular from '../../components/Formular/Formular';

/**
 * Event Page, with Welcome message, next events, etc...
 * @returns Event page component
 */
function EventList(props: { events: any }) {
  const { events } = props;
  console.log(events);

  return <p>Ceci est une liste.</p>;
}

function EventCalendar(props: { events: any }) {
  const { events } = props;
  return (
    <>
      <p>Ceci est un calendrier.</p>
      <Calendar events={events}></Calendar>
    </>
  );
}

function EventView(props: { events: any }) {
  const { events } = props;
  const [value, setValue] = React.useState('1');
  const handleChange = (event: React.SyntheticEvent, newValue: string) => {
    setValue(newValue);
  };
  return (
    <TabContext value={value}>
      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
        <TabList onChange={handleChange} aria-label="lab API tabs example">
          <Tab label="Liste" value="1" />
          <Tab label="Calendrier" value="2" />
        </TabList>
      </Box>
      <TabPanel value="1">
        <EventList events={events}></EventList>
      </TabPanel>
      <TabPanel value="2">
        <EventCalendar events={events}></EventCalendar>
        <CalendarMonth></CalendarMonth>
        <CalendarViewDay></CalendarViewDay>
        <CalendarToday></CalendarToday>
      </TabPanel>
    </TabContext>
  );
}

function Event() {
  const [events, setEvents] = React.useState<Array<EventProps>>([]);

  React.useEffect(() => {
    axios.get('/api/event').then((res: any) => {
      eventsToCamelCase(res.data);
      setEvents(res.data);
    });
  }, []);

  return (
    <>
      <h1>Évènements</h1>
      <p>Ceci est la page des events</p>
      <div style={{ display: 'flex', justifyContent: 'flex-end' }}>
        <Formular />
        <FilterBar />
      </div>
      <EventView events={events} />
    </>
  );
}

export default Event;
