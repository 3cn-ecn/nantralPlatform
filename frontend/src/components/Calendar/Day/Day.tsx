import React from 'react';
import { Grid } from '@mui/material';
import { EventProps } from '../../../Props/Event';
import { EventBlock } from './EventBlock/EventBlock';
import { TimeBlock } from './TimeBlock/TimeBlock';
import './Day.scss';

/**
 * Sort events according to their position.
 * @param eventKeyList The index of events to sort.
 * @param events The global list of events.
 * @returns The list of events sorted.
 */
export function sortWithPos(
  eventKeyList: Array<number>,
  events: Array<EventProps>
): Array<number> {
  const eventChain = [];
  let currentSize = 0;
  let currentEvent = 0;
  let hasBlanked = false;
  let blankUsed = false;
  let blankedEventKey: number;
  const { globalSize } = events[eventKeyList[0]];

  while (
    currentSize !== globalSize &&
    (currentEvent < eventKeyList.length || hasBlanked)
  ) {
    if (currentEvent < eventKeyList.length) {
      // If a blanked event exist, set hasBlanked to true and get the key (= size) of the blanked event
      if (eventKeyList[currentEvent] < 0 && !blankUsed) {
        hasBlanked = true;
        blankedEventKey = currentEvent;
        currentEvent += 1;
      } else if (
        eventKeyList[currentEvent] >= 0 &&
        events[eventKeyList[currentEvent]].position === currentSize
      ) {
        eventChain.push(eventKeyList[currentEvent]);
        currentSize += events[eventKeyList[currentEvent]].effectiveSize;
        currentEvent = 0;
      } else {
        currentEvent += 1;
      }
    } else {
      // Use the blanked event if it's necessary
      eventChain.push(eventKeyList[blankedEventKey]);
      currentSize -= eventKeyList[blankedEventKey];
      currentEvent = 0;
      hasBlanked = false;
      blankUsed = true;
    }
  }
  if (currentSize !== globalSize) {
    throw new Error(`Event chain is wrong`);
  }
  return eventChain;
}

/**
 * The Day component which contains all the TimeBlocks and EventBlocks of this day.
 * @param dayValue The value of the day in the week.
 * @param day The day in the week.
 * @param events The list of events in corresponding day.
 * @param chains The list of blocked chains.
 * @returns The Day component.
 */
export function Day(props: {
  dayValue: number;
  // day: string;
  day: Date;
  events: Array<EventProps>;
  chains: Array<Array<number>>;
}): JSX.Element {
  const { dayValue, day, events, chains } = props;
  const dayChain = [];
  for (let hour = 0; hour < 24; hour++) {
    dayChain.push(<TimeBlock key={hour} startTime={hour}></TimeBlock>);
  }

  // Sort Event chains
  for (let i = 0; i < chains.length; i++) {
    chains[i] = sortWithPos(chains[i], events);
  }

  // Check whether events are already placed
  const toPlace = [];
  for (let i = 0; i < events.length; i++) {
    toPlace.push(i);
  }

  const eventDate = [];

  events.forEach((event) => {
    let startTime = 24;
    if (event.startDate.getDay() === dayValue % 7) {
      if (
        event.startDate.getHours() !== 0 ||
        event.startDate.getMinutes() !== 0 ||
        event.startDate.getSeconds() !== 0
      ) {
        startTime =
          23 -
          event.startDate.getHours() +
          (59 - event.startDate.getMinutes()) / 60 +
          (60 - event.startDate.getSeconds()) / 3600;
      }
    }
    eventDate.push(startTime);
  });

  return (
    <div id={`${day}`} className="blockDisplay">
      <div className="dayData">
        {day.toLocaleDateString('fr-FR', { weekday: 'narrow' })}
      </div>
      <div className="dayData">{day.getDate()}</div>
      {dayChain}
      {chains.map((chain) => (
        <Grid
          container
          key={`Chain${chain}Day${day}`}
          data-testid={`GlobalDayContainer${day}TestId`}
        >
          {chain.map((eventKey) => {
            if (eventKey >= 0) {
              // If the event is not already displayed
              if (toPlace.includes(eventKey)) {
                toPlace.splice(toPlace.indexOf(eventKey), 1);
                return (
                  <Grid
                    item
                    key={events[eventKey].slug}
                    xs={
                      (events[eventKey].effectiveSize * 12) /
                      events[eventKey].globalSize
                    }
                    sx={{
                      height: `0px`,
                      transform: `translate(0px, -${
                        1.2 * eventDate[eventKey]
                      }rem)`,
                    }}
                  >
                    <EventBlock
                      key={events[eventKey].slug}
                      day={dayValue}
                      event={events[eventKey]}
                    />
                  </Grid>
                );
              }

              // If the event has already been displayed
              return (
                <Grid
                  item
                  key={events[eventKey].slug}
                  xs={
                    (events[eventKey].effectiveSize * 12) /
                    events[eventKey].globalSize
                  }
                  sx={{
                    height: `0px`,
                    transform: `translate(0px, -${
                      1.2 * eventDate[eventKey]
                    }rem)`,
                  }}
                ></Grid>
              );
            }

            // return a blanked Event
            return (
              <Grid
                item
                key={eventKey}
                xs={(-eventKey * 12) / events[0].globalSize}
                sx={{
                  height: `0px`,
                }}
              ></Grid>
            );
          })}
        </Grid>
      ))}
    </div>
  );
}
