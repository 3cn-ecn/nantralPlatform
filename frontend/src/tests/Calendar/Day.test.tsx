import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { render } from '@testing-library/react';

import { EventProps } from '../../Props/Event';
import { Day, sortWithPos } from '../../components/Calendar/Day/Day';
import { createTestEvent } from './testElements/testElements';

// Create some events
const events: Array<EventProps> = [];
events.push(
  createTestEvent(
    'The slug1',
    'The test event1',
    new Date('2023-02-11T19:00:00'),
    new Date('2023-02-10T10:22:00'),
    'This is an event to test',
    2,
    6,
    0
  )
);
events.push(
  createTestEvent(
    'The slug2',
    'The test event2',
    new Date('2023-02-10T02:00:00'),
    new Date('2023-02-10T11:38:00'),
    'This is an event to test',
    2,
    6,
    2
  )
);
events.push(
  createTestEvent(
    'The slug3',
    'The test event3',
    new Date('2023-02-10T00:00:00'),
    new Date('2023-02-10T14:54:00'),
    'This is an event to test',
    2,
    6,
    4
  )
);
events.push(
  createTestEvent(
    'The slug4',
    'The test event4',
    new Date('2023-02-10T13:21:00'),
    new Date('2023-02-11T06:56:00'),
    'This is an event to test',
    3,
    6,
    0
  )
);
events.push(
  createTestEvent(
    'The slug5',
    'The test event5',
    new Date('2023-02-10T16:57:00'),
    new Date('2023-02-10T17:22:00'),
    'This is an event to test',
    3,
    6,
    3
  )
);
describe('<Day />', () => {
  it('should display a Day', async () => {
    const component = render(
      <BrowserRouter>
        <Day
          day={new Date('2023-02-10T13:21:00')}
          events={events}
          chains={[
            [0, 1, 2],
            [2, -1, 3],
            [3, 4],
          ]}
        />
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });

  test('sortWithPos should throw an error', async () => {
    expect(() => sortWithPos([2, 3], events)).toThrow(`Event chain is wrong`);
  });
});
