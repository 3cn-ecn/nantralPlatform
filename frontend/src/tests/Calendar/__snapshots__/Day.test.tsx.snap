// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> should display a Day 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="blockDisplay"
        id="Friday, February 10"
      >
        <div
          class="dayData"
        >
          V
        </div>
        <div
          class="dayData"
        >
          10
        </div>
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="timeBlockEven MuiBox-root css-0"
          data-testid="TimeBlockEvenTestId"
        />
        <div
          class="timeBlockOdd MuiBox-root css-0"
          data-testid="TimeBlockOddTestId"
        />
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          data-testid="GlobalDayContainerFriday, February 10TestId"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-milk8-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              style="min-width: 1px; height: 12.44rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <div
                class="imageEventBlur"
              >
                <div
                  class="eventBlockTitleDisplay"
                >
                  The test event1
                </div>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-16tva17-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              style="min-width: 1px; height: 11.559999999999999rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <div
                class="imageEventBlur"
              >
                <div
                  class="eventBlockTitleDisplay"
                >
                  The test event2
                </div>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-milk8-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              style="min-width: 1px; height: 17.88rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <div
                class="imageEventBlur"
              >
                <div
                  class="eventBlockTitleDisplay"
                >
                  The test event3
                </div>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          data-testid="GlobalDayContainerFriday, February 10TestId"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-kv7o3g-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              style="min-width: 1px; height: 12.78rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <div
                class="imageEventBlur"
              >
                <div
                  class="eventBlockTitleDisplay"
                >
                  The test event4
                </div>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-h36ny6-MuiGrid-root"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-milk8-MuiGrid-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          data-testid="GlobalDayContainerFriday, February 10TestId"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-kv7o3g-MuiGrid-root"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-xnvskb-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              style="min-width: 1px; height: 0.5rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <div
                class="imageEventBlur"
              >
                <div
                  class="eventBlockTitleDisplay"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="blockDisplay"
      id="Friday, February 10"
    >
      <div
        class="dayData"
      >
        V
      </div>
      <div
        class="dayData"
      >
        10
      </div>
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="timeBlockEven MuiBox-root css-0"
        data-testid="TimeBlockEvenTestId"
      />
      <div
        class="timeBlockOdd MuiBox-root css-0"
        data-testid="TimeBlockOddTestId"
      />
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        data-testid="GlobalDayContainerFriday, February 10TestId"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-milk8-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            style="min-width: 1px; height: 12.44rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <div
              class="imageEventBlur"
            >
              <div
                class="eventBlockTitleDisplay"
              >
                The test event1
              </div>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-16tva17-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            style="min-width: 1px; height: 11.559999999999999rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <div
              class="imageEventBlur"
            >
              <div
                class="eventBlockTitleDisplay"
              >
                The test event2
              </div>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-milk8-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            style="min-width: 1px; height: 17.88rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <div
              class="imageEventBlur"
            >
              <div
                class="eventBlockTitleDisplay"
              >
                The test event3
              </div>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        data-testid="GlobalDayContainerFriday, February 10TestId"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-kv7o3g-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            style="min-width: 1px; height: 12.78rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <div
              class="imageEventBlur"
            >
              <div
                class="eventBlockTitleDisplay"
              >
                The test event4
              </div>
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-h36ny6-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-milk8-MuiGrid-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        data-testid="GlobalDayContainerFriday, February 10TestId"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-kv7o3g-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-xnvskb-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth eventButton css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
            style="min-width: 1px; height: 0.5rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; vertical-align: top; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <div
              class="imageEventBlur"
            >
              <div
                class="eventBlockTitleDisplay"
              />
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
